phpBB/phpbb/cron/task/core/tidy_search.php

```diff
@@ -114,7 +70,16 @@ public function run()
 	*/
 	public function is_runnable()
 	{
-		return class_exists($this->config['search_type']);
+		try
+		{
+			$this->search_backend_factory->get_active();
```

Comment:
What if you add `$this->search` protected property, then here `$this->search = $this->search_backend_factory->get_active();` and then just `$this->search->tidy()` in the `function run()` above.
Just to avoid `$this->search_backend_factory->get_active();` call duplication.

Reply:
Yes, I've thought about that, but for that it would have to be guaranteed that is_runnable is always called before run.
Also, I personally prefer the service to be stateless.

Reply:
Well, it's always should be called here https://github.com/phpbb/phpbb/blob/master/phpBB/phpbb/cron/event/cron_runner_listener.php#L84 within `is_ready()` function, but ok.

Reply:
I have changed it a little with a solution we both like. Methods don't need to be called in specific order (it works as stateless), and also the call to get_active is done only once.